app/src/main/java/mozilla/lockbox/presenter/RoutePresenter.kt

```diff
@@ -51,23 +51,33 @@ abstract class RoutePresenter(
             return navHostFragmentManager.fragments.lastOrNull()
         }
 
-    private val backListener = OnBackPressedCallback {
-        dispatcher.dispatch(RouteAction.InternalBack)
-        false
+    class BackPressedCallback(
+        val enabled: Boolean = false,
+        val dispatcher: Dispatcher
+    ) : OnBackPressedCallback(enabled) {
+        override fun handleOnBackPressed() {
+            dispatcher.dispatch(RouteAction.InternalBack)
+        }
     }
 
-    private var backListenerCancellable: Cancellable? = null
+    private val onBackPressedDispatcher = activity.onBackPressedDispatcher
+
+    // enabled is defaulted to true. Do we want this?
+    private val callback = BackPressedCallback(false, dispatcher)
```

Comment:
I think `false` is fine; we pass `back` to the system when needed, and let the system handle it when not. I tested this build locally and everything still behaves as I'd expect.